src/dune_pkg/opam_repo.ml

```diff
@@ -378,11 +419,8 @@ let load_all_versions ts opam_package_name =
     | [] -> Fiber.return []
     | packages ->
       let* rev_store = rev_store in
-      let+ opam_files =
-        List.map packages ~f:(fun (repo, file, _) -> repo, file)
-        |> load_packages_from_git rev_store
-      in
-      List.map2 opam_files packages ~f:(fun opam_file (_, _, pkg) -> pkg, opam_file)
+      let+ opam_files = load_packages_from_git rev_store packages in
+      List.map2 opam_files packages ~f:(fun opam_file (_, _, pkg, _) -> pkg, opam_file)
```

Comment:
Maybe it would make sense to change `load_packages_from_git` to save this additional `map2`, given it already does a `map2` over `opam_files`.

Reply:
It's possible but it would be a bit weird. We'd be complicating the type of `load_packages_from_git` by making it "know" the shape the caller wants. Since the perf of this `map2` isn't important, i'd rather leave it simpler.